### PR TITLE
ramips: linkit: use gpiod API

### PR DESCRIPTION
**Pull request by neheb**
gpio.h is deprecated.

ping @hauke @robimarko

### CODE REVIEW

File: target/linux/ramips/patches-6.6/855-linkit_bootstrap.patch [hunk 1]
```diff
@@ -63,13 +61,15 @@
 +	dev_info(&pdev->dev, "Revision : %s\n", hw.rev);
 +
 +	if (!strcmp(hw.board, "LINKITS7688")) {
++		struct gpio_desc *latch;
++
 +		dev_info(&pdev->dev, "setting up bootstrap latch\n");
 +
-+		if (devm_gpio_request(&pdev->dev, LINKIT_LATCH_GPIO, "bootstrap")) {
++		latch = devm_gpiod_get(&pdev->dev, "bootstrap", GPIOD_OUT_LOW);
```
Comment (DragonBluep) [inline target/linux/ramips/patches-6.6/855-linkit_bootstrap.patch:68]: I think this won't work if we don't describe `bootstrap` in device tree.

File: target/linux/ramips/patches-6.6/855-linkit_bootstrap.patch [hunk 2]
```diff
@@ -63,13 +61,15 @@
 +	dev_info(&pdev->dev, "Revision : %s\n", hw.rev);
 +
 +	if (!strcmp(hw.board, "LINKITS7688")) {
```
Comment (DragonBluep) [inline target/linux/ramips/patches-6.6/855-linkit_bootstrap.patch:63]: I suggest removing this patch. We can control this bootstrap pin (GPIO#11) in device tree via gpio-export or gpio-hog. BTW, I don't see this GPIO on Pin-out diagram. It's a hack for custom modified hardware?
image
image

Comment (neheb) [reply target/linux/ramips/patches-6.6/855-linkit_bootstrap.patch:63]: Got it.

Comment (DragonBluep) [reply target/linux/ramips/patches-6.6/855-linkit_bootstrap.patch:63]: I think you can change the title and make this change here if you like.

Comment (neheb) [reply target/linux/ramips/patches-6.6/855-linkit_bootstrap.patch:63]: Something like this?

```
--- a/target/linux/ramips/dts/mt7628an_mediatek_linkit-smart-7688.dts
+++ b/target/linux/ramips/dts/mt7628an_mediatek_linkit-smart-7688.dts
@@ -22,10 +22,16 @@
                serial0 = &uart2;
        };
 
-       bootstrap {
-               compatible = "mediatek,linkit";
+       export {
+               compatible = "gpio-export";
+               #size-cells = <0>;
+
+               bootstrap {
+                       gpio-export,name = "bootstrap";
+                       gpio-export,output = <1>;
+                       gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+               };
 
-               status = "okay";
        };
 
        leds {

```

Comment (DragonBluep) [reply target/linux/ramips/patches-6.6/855-linkit_bootstrap.patch:63]: Yes. The gpio-export should be fine.

Finally I know, this pin (GPIO#11, aka GPIO 0) is used to control the bus switch chip. Output low level will enable export i2s, spi and uart bus. ref: https://seeeddoc.github.io/LinkIt_Smart_7688/res/Hardware_Schematics.zip
image

Comment (neheb) [reply target/linux/ramips/patches-6.6/855-linkit_bootstrap.patch:63]: Maybe gpio-hog is better. Not sure. I don't know if this really needs to be controlled.

**Pull request closed without merging**